### ISSUE TITLE
Allow the user to send extra metadata with each transaction

### ISSUE THREAD

**Issue by hackaugusto** [ethereum/pyethapp#159]
For a testing environment with lots of transactions, if one transaction fails it can be hard to figure out what was it, allowing the user to send extra metada data in the rpc client and server can be quite useful for debugging.

**Comment by RomanZacharia** [comment]
@hackaugusto  So the request is just to print the additional metadata to the console when the transaction fails, right?

**Comment by hackaugusto** [comment]
Not really. First we need to decide if this is worth doing, since it is an incompatible change to the rpc server, then how the user can provide additional metada data, it could be provided at the contract creation or at with a transaction, then priting the values is possible.
